womtool/src/main/scala/womtool/WomtoolMain.scala

```diff
@@ -75,25 +75,44 @@ object WomtoolMain extends App {
   }
 
   def upgrade(workflowSourcePath: String): Termination = {
+    import wdl.model.draft3.elements.ImportElement
+    import wdl.draft2.model.Import
+    import languages.wdl.draft2.WdlDraft2LanguageFactory.httpResolver
+
+    // Get imports directly from WdlNamespace, because they are erased during WOMification
+    val maybeWdlNamespace: Try[WdlNamespace] =
+      // TODO: do we need directory resolver also? Not clear how to reference it.
+      WdlNamespace.loadUsingPath(Paths.get(workflowSourcePath), None, Some(List(WdlNamespace.fileResolver, httpResolver)))
+
+    def upgradeImport(draft2Import: Import): ImportElement = {
+      if (draft2Import.namespaceName.nonEmpty)
+        // draft-2 does not have structs, so the source WDL will not have any for us to rename
+        ImportElement(draft2Import.uri, Some(draft2Import.namespaceName), Map())
```

Comment:
Switch to `Option.apply`.

If `namespaceName` somehow ends up `null`, this is going to be a world a pain to debug.

Reply:
TIL and I agree 100%

![screen shot 2018-07-30 at 11 14 39 am](https://user-images.githubusercontent.com/1087943/43406241-d4486246-93e9-11e8-8787-c5d8204d77b7.png)

Reply:
IMO "quick failure" is better than "silently pretending it was fine" in those cases where non-optional values somehow end up `null`, though I'm in a minority on this one... :)